org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/findrefs/ReferenceQuery.java

```diff
@@ -75,7 +76,14 @@ public ISearchResult getSearchResult() {
 	@Override
 	public IStatus run(IProgressMonitor monitor) throws OperationCanceledException {
 		searchResult.reset();
-		finder.findAllReferences(targetURIs, localContextProvider, searchResult, monitor);
+		try {
+			finder.findAllReferences(targetURIs, localContextProvider, searchResult, monitor);
+		} catch (WrappedException e) {
+			if (e.getCause() instanceof OperationCanceledException) {
```

Comment:
Maybe add `|| monitor.isCancelled()` to the condition and simply throw a new OCE?

Reply:
thought of that but that would cover other exceptions if cancelling happens after exception and before catch

Reply:
I'm not sure that the OCE will be wrapped into a single WrappedException all the time. Do we need the call to finish in case of errors, too?

Reply:
youn mean catch and log the rest? obviously the caller catches all bad exceptions and log them so it looks good to me to rethrow. maybe we can use the cause finder from guava?

Reply:
:+1: